Update 1wf_01_debugging.md (#2897)
ocaml/ocaml.org: data/tutorials/guides/1wf_01_debugging.md

be output automatically by the generic pretty-printer used by the trace
mechanism.

> Compiler builtins improve debugging messages. The location of the error in the program can be displayed. 

Reviewer: ```suggestion
Compiler builtins help display useful debugging messages. They indicate a location in the program's source. 
```

For example,
```ocaml
match Message.unpack response with
| Some y -> y
| None -> (Printf.eprintf "Invalid message at %s" __LOC__; raise Invalid_argument)
```

> At compile time, the `__LOC__` builtin is substituted with its location in the program, described as a string `"File %S, line %d, characters %d-%d"`. One can also get the file name, line number, start character and end character directly through the `__POS__` builtin:

Reviewer: ```suggestion
At compile time, the `__LOC__` builtin is substituted with its location in the program, described as a string `"File %S, line %d, characters %d-%d"`. File name, line number, start character and end character are also available through the `__POS__` builtin:
```

be output automatically by the generic pretty-printer used by the trace
mechanism.

> Print debugging is assisted by compiler builtins that allow a printed error 
> message to refer to the program location that the error was raised from. 

Reviewer: ```suggestion
Compiler builtins improve debugging messages. The location of the error in the program can be displayed. 
```

For example,
```ocaml
match Message.unpack response with
| Some y -> y

> | None -> Printf.printf "Invalid message at %s" __LOC__; flush stdout;
>           raise Invalid_argument

Reviewer: ```suggestion
| None -> (Printf.eprintf "Invalid message at %s" __LOC__; raise Invalid_argument)
```

```

> The compiler builtin `__LOC__` is substituted at compile time with the 
> location it occurs in the program, described as a string 
> "File %S, line %d, characters %d-%d".
> One can also get the file name, line number, start character and
> end character directly through the __POS__ builtin:

Reviewer: ```suggestion
At compile time, the `__LOC__` builtin is substituted with its location in the program, described as a string `"File %S, line %d, characters %d-%d"`. One can also get the file name, line number, start character and end character directly through the `__POS__` builtin:
```

```ocaml
match Message.unpack response with
| Some y -> y
| None ->
    let fname, lnum, _cstart, _cend = __POS__ in
    Printf.printf "At line %d in file %s, an incorrect response was passed to Message.unpack"
    lnum fname;
    flush stdout; raise Invalid_argument
```

> Documentation for all debugging compiler builtins is in the 
> [standard library documentation](https://ocaml.org/manual/5.2/api/Stdlib.html#1_Debugging).

Reviewer: ```suggestion
Compiler builtins are described in the
[standard library](https://ocaml.org/manual/5.2/api/Stdlib.html#1_Debugging) documentation.
```
